src/server/builder.rs

```diff
@@ -0,0 +1,181 @@
+            #[cfg(feature = "swagger")]
+            swagger_mount: "/swagger",
+            #[cfg(feature = "swagger")]
+            swagger_spec: DefaultApiRaw::default(),
+        }
+    }
+}
+
+impl<'a> Microservice<'a>
+{
+    pub fn set_app_port(mut self, app_port: &'a str) -> Self
+    {
+        self.app_port = app_port;
+        self
+    }
+
+    pub fn set_run_env(mut self, run_env: &'a str) -> Self
+    {
+        self.run_env = run_env;
+        self
+    }
+
+    #[cfg(feature = "swagger")]
+    pub fn set_swagger_mount(mut self, swagger_mount: &'a str) -> Self
+    {
+        self.swagger_mount = swagger_mount;
+        self
+    }
+
+    #[cfg(feature = "swagger")]
+    pub fn set_swagger_spec(mut self, swagger_spec: DefaultApiRaw) -> Self
+    {
+        self.swagger_spec = swagger_spec;
+        self
+    }
+
+    #[cfg(feature = "swagger")]
+    pub fn set_swagger_info(mut self, pkg_name: impl Into<String>, pkg_version: impl Into<String>, pkg_description: impl Into<String>) -> Self
+    {
+        self.swagger_spec.info.title = pkg_name.into();
+        self.swagger_spec.info.version = pkg_version.into();
+        self.swagger_spec.info.description = Some(pkg_description.into());
+        self
+    }
+
+    pub async fn start<D, T, F, C> (
+        self,
+        prepare_app_data: impl Fn() -> T + Sized,
+        configure_app: F,
+        cors_factory: Option<C>,
+    ) -> std::io::Result<()>
+    where
+        D: AppDataWrapper + 'static,
+        T: StatsPresenter<D> + 'static + Clone + Send + Sync,
+        F: Fn(&mut web::ServiceConfig) + Send + Clone + 'static + Sized,
+        C: Fn() -> Cors + Send + Clone + 'static,
+    {
+        if let Some(cors_factory) = cors_factory {
+            self.start_inner(prepare_app_data, configure_app, cors_factory).await
+        } else {
+            self.start_inner(prepare_app_data, configure_app, default_cors).await
+        }
+    }
+
+    async fn start_inner<D, T, F, C> (
+        self,
+        prepare_app_data: impl Fn() -> T + Sized,
+        configure_app: F,
+        cors_factory: C,
+    ) -> std::io::Result<()>
+    where
+        D: AppDataWrapper + 'static,
+        T: StatsPresenter<D> + 'static + Clone + Send + Sync,
+        F: Fn(&mut web::ServiceConfig) + Send + Clone + 'static + Sized,
+        C: Fn() -> Cors + Send + Clone + 'static,
+    {
+        dotenv().ok();
+
+        match logger::init_logger() {
+            Ok(_) => info!("Logger has been initialized"),
+            Err(_) => error!("Error logger initialization")
+        };
+
+        //env::set_var("RUST_LOG", "actix_web=debug");
+
+        let numthreads = threads::num_threads();
+        info!("Configuring for {} threads", numthreads);
+
+        let app_data = web::Data::new(prepare_app_data());
+        let stats = web::Data::new(BaseStats::default());
+
+        #[allow(unused)]
+        let prod_env = self.run_env == "prod";
+
+        let app_port = self.app_port.to_string();
```

Comment:
if we are allocating string here - why not just keep `app_port: String` in builder? For ease of use `set_app_port` could accept `ToString` to allow different types etc

Same for other &str fields

Reply:
This will increase string allocations as default() will at once alloc 3 strings. I just removed this `to_string()` :-)